
# **Stalkie Reference Genome Assembly**
* This pipeline _will_ generate reference genomes for _Teleopsis dalmanni_, _Teleopsis whitei_ and _Diasemopsis meigenii_. PacBio HiFi data for assembly, Omni-C for Scaffolding, and RNA ISO-seq for annotation.
From [Alex's pipeline](https://github.com/alexjvr1/T.dalmanni_Genomics_of_meiotic_drive/blob/main/Electronic_Lab_Book.md#1-genome-assembly), we know the current _T. dalmanni_ ref is ~0.383 Gb and 3 chromosomes 

### Samples & data for assembly
  ~~~

  - Assemblies should be generated for each sex, so 6 ssemblies total, enabling sex chromosome identification
  - HiFi and HI-C data for 1M and 1F per species
  - 5 illumina DNA-seq per species 
  ~~~
# 0. **Raw Read QC**

# 1. **Genome Size Estimation with Jellyfish**
* [Jellyfish v2.2.10](https://github.com/gmarcais/Jellyfish)
* Ran [jellyfish.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/jellyfish/jellyfish.sh) on each species
* Used [genomescope](http://qb.cshl.edu/genomescope/) to visualise the .histo files generated by jellyfish (read length=10000, max kmer coverage=100000)

# 2. **Assembly with Hifiasm**
* [hifiasm v0.16.1-r375](https://github.com/chhylp123/hifiasm)
* Ran [hifiasm.sh]([https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/hifiasm.sh](https://github.coý/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/hifiasm/hifiasm.sh)
* This uses both hi-c and hifi reads

## **Checking Coverage is as Expected**
* Check hifiasm correctly identifies the homozygous and heterozygous peaks in the kmer plots, compare this to the jellyfish kmer plots as a sanity check.
  - This is true for meigenii_4, dalmanni_7
* Issue with meigenii_5 and dalmanni_6:  "peak_hom: 23; peak_het: -1"
  - [see this thread](https://github.com/chhylp123/hifiasm/issues/245), potentially caused by the sample being highly heterozygous. Hifiasm only identifies one peak (the het peak) and misidentifies it as the hom peak. For meigenii_5, it should be "peak_het: 23, peak_hom: 46"
  - This can be fixed by specifying -k 19 (kmer length, default 51, must be <64)
  - this fix makes the script take ages to run, and for some samples may require too much memory, also [this person](https://github.com/chhylp123/hifiasm/issues/55) found that doing this made hifiasm output both haplotypes into the primary assembly, leaving a too small alternate assembly - make sure it isn't doing this.
  - Also no indication as to why they chose this kmer length 

# 3. **Scaffolding with YAHS**
gfa to fasta
~~~bash
awk '/^S/{print ">"$2"\n"$3}' $input_file | fold > $output_file
~~~

* Mapping hic reads with the arima pipeline (one of the ones recomended by yahs): [1_mapping](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/scaffolding/yahs/1_mapping.sh)
* Then run [2_yahs.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/scaffolding/yahs/2_yahs.sh)
* This will output a scaffolded assembly and juicer input files for manual curation

## **BUSCO**
* Ran [busco.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/utility/busco.sh)
* Looks good, except female dalmanni is highl duplicated

# **4. polishing**

## **filter contanimants with blobtoolkit**
* [Blobtoolkit v3.1.6 ](https://github.com/blobtoolkit/blobtoolkit)
* Essentially takes blast output, coverage, and busco output, and filters the dataset for contamination
* Needs:
  - blast info (generated in [blast_par.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/blobtoolkit/blast_par.sh))
  - coverage info (generated in [blobtoolkit.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/blobtoolkit/blobtoolkit.sh))

### **Blast**
**Installing local blast database**
* Remote searches are not feasible for large queries, so installed the local Blast nt db ([using blast_nt_db.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/blobtoolkit/blast_nt_db.sh)), this takes up ~450GB of space
* Even with the local blast nt db, blasting my highly contiguous assemblies will take too long unless it is parallelised

**Parallelising Blast**
* Split assembly fasta into 500 smaller files: ran [fasta_splitter.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/blobtoolkit/fasta_splitter.sh)
* ran blast on each using the script, [blast_par.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/blobtoolkit/blast_par.sh)
* Then I condensed these blast output files into a single file:
~~~
cat out/* >> all_blast.out
~~~

## **Blobtoolkit pipeline:**
* Ran [blobtoolkit.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/edit/main/scripts/blobtoolkit/blobtoolkit.sh)
* This script calculates coverage, takes the blast output I just generated, and the busco output from [busco.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/edit/main/scripts/utility/busco.sh). It then filters the assembly for contamination, I am able to specify which taxanomic rank blast matches I want removed.
* **Important: generating coverage using minimap2 does not work for dalmanni_6, this is likely an issue to to with the HiFi reads being to long.**
* This analysis has currently been run on whitei_1 and whitei_2 only.

#### **Setting Filter Parameters**
* Need to select appropriate taxanomic level to filter
* The below table was generated with: --param bestsum_$clade_rank--keys=no-hit,$clade # this keeps only reads with blast hits from that clade, and reads with no blast hits

| Clade      | Size (Gb) | Contigs | Completeness | Duplication|
|------------|-----------|---------|--------------|------------|
| Unfiltered | 0.762     | 7598    | 95.8         | 35.6       |
| Eukaryota  | 0.748     |  7481   | 95.7         | 35.6       |
| Metazoa    | 0.742     | 7368    | 95.6         | 35.5       |
| Arthropoda | 0.742     | 7363    | 95.6         | 35.5       |
| Diptera    | 0.715     | 6933    | 95.3         | 35.3       | 

* Metazoan and no blast matches seem to be the most sensible contigs to retain.

## **Removing Dups with Purge Dups**
* [purge_dups v1.2.5](https://github.com/dfguan/purge_dups)
* ran [purge_dups_all.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/purge_dups/purge_dups_all.sh)
* Purges haplotigs using coverage and sequence similarity info



