
# **Reference Genome Assembly**

[Useful resource](https://github.com/alexjvr1/T.dalmanni_Genomics_of_meiotic_drive/blob/main/Electronic_Lab_Book.md#1-genome-assembly)

## **1. Genome Size Estimation with Jellyfish**
* memory decided via hash size (size of the genome plus seq error) T. dalmanni genome is 0.62 gb, so i'll assume 0.7 (nanoplot doesnt accept less than 1G so do that)
* not sure what kmer size to use, 21 seems sensible 
* jellyfish V 2.2.10

Ran [jellyfish.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/jellyfish.sh) on each species

### **kmer distribution plots**
Used [genomescope](http://qb.cshl.edu/genomescope/) to visualise the .histo files generated by jellyfish (read length=10000, max kmer coverage=100000):
- [_T. whitei_ Female](http://genomescope.org/analysis.php?code=blR2SdZ6dlrwedM2Fgs8)
- [_T. whitei_ Female Driver](http://genomescope.org/analysis.php?code=VjP3GGnGS6YEajK6XUB1)
- [_T. whitei_ Male](http://genomescope.org/analysis.php?code=XWsvvvcObVy7x83qxEAu)
- [_T. dalmanni_ Female](http://genomescope.org/analysis.php?code=7NAz3STmNnNqW83R5kLc)
- [_T. dalmanni_ Male](http://genomescope.org/analysis.php?code=TW147FQXJ18A79EpLjEc)
- [_D. meigenii_ Female](http://genomescope.org/analysis.php?code=4tooUMZucAlqGmV0V0E8)
- [_D. meigenii Male_](http://genomescope.org/analysis.php?code=ISWwt723ucFqtJJKRgrT) 
<br><br>

| file prefix | sample         |     |
| ------------ | --------------- | ------ |
| 200437\_1    | whitei_F        |
| 200437\_2    | whitei_M        | Male   |
| 200437\_3    | whitei_F_driver | Female |
| 200437\_4    | meigenii_F      | Female |
| 200437\_5    | meigenii_M      | Male   |
| 200437\_6    | dalmanni_M     | Male   |
| 200437\_7    | dalmanni_F     | Female |

## **2. Assembly with Hifiasm**
* [Documentation](https://github.com/chhylp123/hifiasm)
* Ran [hifiasm.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/hifiasm.sh) (Hifiasm V 0.16.1), takes ~10-12 hrs

### Samples:

  ~~~
  - 7 total, 2 per species (1M, 1F), plus 1 additional T. whitei F (potential driver).
  - Assemblies should be generated for each sample, so 7 assemblies total.
  - Two HiFi .fastq.gz files per individual (so 4,4,6) per species.
  ~~~

| Sample ID | file\_prefix | Species         | Sex    |
| --------- | ------------ | --------------- | ------ |
| ST1       | 200437\_1    | C.whitei        | Female |
| ST2       | 200437\_2    | C.whitei        | Male   |
| ST3       | 200437\_3    | C.whitei Driver | Female |
| ST6       | 200437\_4    | D.meigenii      | Female |
| ST7       | 200437\_5    | D.meigenii      | Male   |
| ST8       | 200437\_6    | T.dalmanni      | Male   |
| ST9       | 200437\_7    | T.dalmanni      | Female |

### **Hifiasm output**
* [output files documentation](https://hifiasm.readthedocs.io/en/latest/interpreting-output.html)
* duplication looks very high across the board, not very contiguous

### **Inital QC with BUSCO**
* Used a docker/apptainer image
* ran [busco.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/busco.sh), takes ~10-20 mins
* see [busco output](https://github.com/BenAlston/stalkie_ref_genome_assembly/tree/main/lab_book/Data/BUSCO_output)
* very high duplication in T. dalmanni M, see if this is consistent across all species

### **Alignment With Miniap2**
* [Documentation](https://github.com/lh3/minimap2)

| Sample ID | aligned | Species         | Sex    |
| --------- | ------------ | --------------- | ------ |
| 1       | y    | C.whitei        | Female |
| 2       | n    | C.whitei        | Male   |
| 3       | n    | C.whitei Driver | Female |
| 4       | y    | D.meigenii      | Female |
| 5       | n    | D.meigenii      | Male   |
| 6       | y    | T.dalmanni      | Male   |
| 7       | n    | T.dalmanni      | Female |



### **Filtering with Blobtoolkit**
* [documentation](https://github.com/blobtoolkit/blobtoolkit)
* make a blobdir: containing the .fa file (symlinked), and a metadata .yaml file (optional)
* Blobtools has been a pain to get working, I have been able to filter dalmanni M, which makes the assembly more contiguous by ~500, and n50 has increased from 600kb to almost 1mb. The genome is around 90 mb shorter now.
* however, im not sure i have done everything correctly, going to chat with noah about this next week.
* For now, try and get megablast working, then focus on other stuff
#### Blast
* Remote searches take too long, so installing the Blast nt db

### **Alternatives to Blobtoolkit**
* [FCS-GX](https://github.com/ncbi/fcs/wiki/FCS-GX) is NCBIs contaminant removal tool. Looks good but requires installing a 470GB reference dataset.

## **Purge Dups**
* [Documentation](https://github.com/dfguan/purge_dups)
* First running this pipeline on whitei F (sample 1) since its busco dup is 35%


## **Next Steps:**
*  [findZX](https://github.com/hsigeman/findZX) has potential, look through the paper


1. Get blobtools working
3. use inspector for qc
4. extract mt genomes
6. running jellyfish again doesn't work, try harder to make it work
7. use samtools to determine proportion of reads that map
8. map short-read data
9. use samtools to look at coverage




