
# **Reference Genome Assembly**

[Useful resource](https://github.com/alexjvr1/T.dalmanni_Genomics_of_meiotic_drive/blob/main/Electronic_Lab_Book.md#1-genome-assembly)

Current _T. dalmanni_ ref is ~0.383 Gb and 3 chromosomes

## **1. Genome Size Estimation with Jellyfish**
* memory decided via hash size (size of the genome plus seq error) wilkinson T. dalmanni genome is 0.62 gb (edit: did not have access to the pom one at this stage) , so i'll assume 0.7 (nanoplot doesnt accept less than 1G so do that)
* not sure what kmer size to use, 21 seems sensible 
* jellyfish V 2.2.10

Ran [jellyfish.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/jellyfish.sh) on each species

### **kmer distribution plots**
Used [genomescope](http://qb.cshl.edu/genomescope/) to visualise the .histo files generated by jellyfish (read length=10000, max kmer coverage=100000):
- [_T. whitei_ Female](http://genomescope.org/analysis.php?code=blR2SdZ6dlrwedM2Fgs8)
- [_T. whitei_ Female Driver](http://genomescope.org/analysis.php?code=VjP3GGnGS6YEajK6XUB1)
- [_T. whitei_ Male](http://genomescope.org/analysis.php?code=XWsvvvcObVy7x83qxEAu)
- [_T. dalmanni_ Female](http://genomescope.org/analysis.php?code=7NAz3STmNnNqW83R5kLc)
- [_T. dalmanni_ Male](http://genomescope.org/analysis.php?code=TW147FQXJ18A79EpLjEc)
- [_D. meigenii_ Female](http://genomescope.org/analysis.php?code=4tooUMZucAlqGmV0V0E8)
- [_D. meigenii Male_](http://genomescope.org/analysis.php?code=ISWwt723ucFqtJJKRgrT) 
<br><br>

## **2. Assembly with Hifiasm**
* [Documentation](https://github.com/chhylp123/hifiasm)
* Ran [hifiasm.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/hifiasm.sh) (Hifiasm V 0.16.1), takes ~10-12 hrs

### Samples:

  ~~~
  - 7 total, 2 per species (1M, 1F), plus 1 additional T. whitei F (potential driver).
  - Assemblies should be generated for each sample, so 7 assemblies total.
  - Two HiFi .fastq.gz files per individual (so 4,4,6) per species.
  ~~~

| Sample ID | file\_prefix | Species         | Sex    |
| --------- | ------------ | --------------- | ------ |
| ST1       | 200437\_1    | C.whitei        | Female |
| ST2       | 200437\_2    | C.whitei        | Male   |
| ST3       | 200437\_3    | C.whitei Driver | Female |
| ST6       | 200437\_4    | D.meigenii      | Female |
| ST7       | 200437\_5    | D.meigenii      | Male   |
| ST8       | 200437\_6    | T.dalmanni      | Male   |
| ST9       | 200437\_7    | T.dalmanni      | Female |

### **Hifiasm output**
* [output files documentation](https://hifiasm.readthedocs.io/en/latest/interpreting-output.html)
* duplication looks very high across the board, not very contiguous


|sample| info         | est_size         | primary_assembly_size | busco_completeness | busco_dup | contigs | N50 (Kb) |
|-----|-----------------|------------------|-----------------------|--------------------|-----------|---------|-----|
| whitei_1 | female       |        597436869 | 761502157             | 95.8               | 35.6      | 7589    | 156 |
| whitei_2 | male       | 424970908        | 730392625             | 95.1               | 25.6      | 6466    | 178 |
| whitei_3 | female,potential driver |        584931232 | 731617842             | 95.7               | 42.3      | 6774    | 178 |
| meigenii_4 | female      |        694112101 | 694692253             | 96.4               | 28.6      | 3136    | 463 |
| meigenii_5 | male     | 477607885        | 742331081             | 96.3               | 16.9      | 3064    | 576 |
|dalmanni_6 | male       | 657136228        | 700361654             | 97.8               | 25.7      | 2913    | 687 |
| dalmanni_7 | female       |        533280821 | 609130602             | 97.3               | 4.9       | 2425    | 998 |




### **BUSCO**
* Used a docker/apptainer image
* ran [busco.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/busco.sh), takes ~10-20 mins
* see [busco output](https://github.com/BenAlston/stalkie_ref_genome_assembly/tree/main/lab_book/Data/BUSCO_output)
* very high duplication across all species, completeness is ok

### **Coverage**
* genomescope plots are no longer working, will need redoing
* Hifiasm outputs have been misplaced, rerruning for meigenii and make sure it's all working, saving the outputs somewhere sensible this time


### **Alignment With Miniap2**
* [Documentation](https://github.com/lh3/minimap2)

| Sample ID | aligned | Species         | Sex    |
| --------- | ------------ | --------------- | ------ |
| 1       | y    | C.whitei        | Female |
| 2       | n    | C.whitei        | Male   |
| 3       | n    | C.whitei Driver | Female |
| 4       | y    | D.meigenii      | Female |
| 5       | n    | D.meigenii      | Male   |
| 6       | y    | T.dalmanni      | Male   |
| 7       | n    | T.dalmanni      | Female |



## **Filtering with Blobtoolkit**
* [documentation](https://github.com/blobtoolkit/blobtoolkit)
* essentially takes blast output, coverage, and busco output, and filters the dataset for contamination. This program has been quite tricky to get working, considering it isn't doing anything that complext. This is probably because it does a bunch of other stuff that isn't relevant for what I'm doing.

### **Blast**
* Needed for the filtering stage of btk
* Remote searches take too long, so installing the Blast nt db
* only 500gb, installed using docker image [blast_nt_bd.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/tree/main/scripts/blast_nt_db.sh)
* Script ran overnight, has takes around an hour per seq, so would take tens of days for all 2400 contigs
* need to parralelize

* split fasta into 500 smaller files: ran [fasta_splitter.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/fasta_splitter.sh)
* ran blast on each using an array script, [blast_par.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/tree/main/scripts/blast_par.sh)

then condense into a single blast output file and remove temp files:
~~~
# when blast script has run, concentrate all outputs into one file
cat out/* >> all_blast.out
# should also write a script to remove the temp files
~~~

### **Blobtoolkit pipeline:**
* ran [btk.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/edit/main/scripts/btk.sh) (work in progress)
* **Add BUSCO** - done
* **Add Blast Hits** - done for dalmanni and whitei
* **Add coverage** (long read remapped to primary assembly) - done for whitei_1
  -  does not work for dalmanni_6, ruled out corrupted .fa, could be a complex issue with this sample, not going to worry for now, going to proceed using whitei_2 instead
* **adding blast hits** - Done for whitei_2, but will need reworking to run efficiently on all samples
* filtering by blast hits now works

#### **Setting Filter Parameters**
* Need to select taxanomic level
* The below table was generated with: --param bestsum_$clade_rank--keys=no-hit,$clade # this keeps only reads with blast hits from that clade, and reads with no blast hits

| Clade      | Size (Gb) | Contigs | Completeness | Duplication|
|------------|-----------|---------|--------------|------------|
| Unfiltered | 0.762     | 7598    | 95.8         | 35.6       |
| Eukaryota  | 0.748     |  7481   | 95.7         | 35.6       |
| Metazoa    | 0.
| Arthropoda | 0.742     | 7363    | 95.6         | 35.5       |
| Diptera    | 0.715     | 6933    | 95.3         | 35.3       | 

* mannually looked at filtered contigs (both those retained and removed), looks like blobtools is doing what it is supposed to


**Summary:**
* Can't add coverage for Dalmanni_6
* ran [blobtoolkit.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/new/main/scripts/blobtoolkit.sh), filtered the assembly to retain only metazoan and no blast match reads

### **Alternatives to Blobtoolkit**
* [FCS-GX](https://github.com/ncbi/fcs/wiki/FCS-GX) is NCBIs contaminant removal tool. Looks good but requires installing a 470GB reference dataset.

## **Purge Dups**
* [Documentation](https://github.com/dfguan/purge_dups)
* First running this pipeline on whitei F (sample 1) since its busco dup is 35%
* generate config dir, ran [purge_dups_config.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/tree/main/scripts/purge_dups_config.sh)
  - seems to work, no error or output
  - manually modified the 1_config.json file to increase available memory (uses minimap2 so should need quite a bit)

* Got it to run, although the automatic busco doesnt work for some reason, and the output purged assembly has very clearly gone horribly wrong (40% completeness). It would be nice if the output was explained in the documentation.
  - there dont seem to be issues with the data going in, so i probably need to tweak the parameters until the output looks good, to do this i'll need to set up the pipeline manually (instead of just running run_purge_dups.py on the config dir) this will take time

**Manual pipeline**
* Issues generating .paf files
* running paf_gen.sh - works
* ran the pipeline, on whitei_1

whitei_1 pre and post purge_dups_manual.sh
|              | length (gb) | contigs | dup (%)| Conpleteness (%) | n50 (kb) |
| -------------|-------------|---------|--------|------------------|----------|
| pre-purging  | 0.762 | 7589   |  35.6|   95.8         | 156 |
| purge_dups round 1  | 0.475    | 4653      | 3.3    | 90.9     | 161 |
| purge_dups round 2  | 0.542    | 5156      | 1.2    | 95.4     | 177 |

in purge_dups_man_1 & 2.sh, the line: 'minimap2 -xasm5 -DP ${REF}.split ${REF}.split | gzip -c - > $(basename $REF).split.self.paf.gz', check -xasm5 is correct here
* need to make sure purge_dups is doing everything correctly
* set up the script to run on all sampes
* running job  2488718
## **Next Steps:**
*  [findZX](https://github.com/hsigeman/findZX) has potential, look through the paper


4. extract mt genomes
7. use samtools to determine proportion of reads that map
8. map short-read data
9. use samtools to look at coverage




