
# **Reference Genome Assembly**

[Useful resource](https://github.com/alexjvr1/T.dalmanni_Genomics_of_meiotic_drive/blob/main/Electronic_Lab_Book.md#1-genome-assembly)

## **1. Genome Size Estimation with Jellyfish**
* memory decided via hash size (size of the genome plus seq error) T. dalmanni genome is 0.62 gb, so i'll assume 0.7 (nanoplot doesnt accept less than 1G so do that)
* not sure what kmer size to use, 21 seems sensible 
* jellyfish V 2.2.10

Ran [jellyfish.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/jellyfish.sh) on each species

### **kmer distribution plots**
Used [genomescope](http://qb.cshl.edu/genomescope/) to visualise the .histo files generated by jellyfish (read length=10000, max kmer coverage=100000):
- [_T. whitei_ Female](http://genomescope.org/analysis.php?code=blR2SdZ6dlrwedM2Fgs8)
- [_T. whitei_ Female Driver](http://genomescope.org/analysis.php?code=VjP3GGnGS6YEajK6XUB1)
- [_T. whitei_ Male](http://genomescope.org/analysis.php?code=XWsvvvcObVy7x83qxEAu)
- [_T. dalmanni_ Female](http://genomescope.org/analysis.php?code=7NAz3STmNnNqW83R5kLc)
- [_T. dalmanni_ Male](http://genomescope.org/analysis.php?code=TW147FQXJ18A79EpLjEc)
- [_D. meigenii_ Female](http://genomescope.org/analysis.php?code=4tooUMZucAlqGmV0V0E8)
- [_D. meigenii Male_](http://genomescope.org/analysis.php?code=ISWwt723ucFqtJJKRgrT) 
<br><br>
* Some seem acceptable, containing two peaks, others e.g., _D. meigenii_ F have less distinct peaks, this should still be fine
* However, est. genome size seems odd, being variable, and larger in females than males in _D. meigenii_ and _T. whitei_ and the opposite in _T. dalmanni_

## **2. Assembly with Hifiasm**
* [Documentation](https://github.com/chhylp123/hifiasm)
* Ran [hifiasm.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/hifiasm.sh) (Hifiasm V 0.16.1), takes ~10-12 hrs

### Samples:

  ~~~
  - 7 total, 2 per species (1M, 1F), plus 1 additional T. whitei F (potential driver).
  - Assemblies should be generated for each sample, so 7 assemblies total.
  - Two HiFi .fastq.gz files per individual (so 4,4,6) per species.
  ~~~

| Sample ID | file\_prefix | Species         | Sex    |
| --------- | ------------ | --------------- | ------ |
| ST1       | 200437\_1    | C.whitei        | Female |
| ST2       | 200437\_2    | C.whitei        | Male   |
| ST3       | 200437\_3    | C.whitei Driver | Female |
| ST6       | 200437\_4    | D.meigenii      | Female |
| ST7       | 200437\_5    | D.meigenii      | Male   |
| ST8       | 200437\_6    | T.dalmanni      | Male   |
| ST9       | 200437\_7    | T.dalmanni      | Female |

### **Hifiasm output**
* [output files documentation](https://hifiasm.readthedocs.io/en/latest/interpreting-output.html)

### **Used Quast for basic summary stats**
* ran [quast.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/quast.sh) on primary assembly for dalmanni_7
* will probably add this script directly to the hifiasm one for the main assembly step
* N50 of 600Mb is not great
* busco does this so actually not needed

### **Inital QC with BUSCO**
* Had to use a docker/apptainer image for this
* ran [busco.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/busco.sh), takes ~10-20 mins
* see [busco output](https://github.com/BenAlston/stalkie_ref_genome_assembly/tree/main/lab_book/Data/BUSCO_output)
* very high duplication in T. dalmanni M, see if this is consistent across all species


### **Filtering with Blobtools2**
* [documentation](https://github.com/blobtoolkit/blobtoolkit)


## **Next Steps:**
*  [findZX](https://github.com/hsigeman/findZX) has potential, look through the paper

1. run hifiasm on whitei and meigenii - _currently running_
2. use blobtools to remove contamination 
3. use inspector for qc
4. extract mt genomes
5. remap reads to genome
6. run jellyfish again
7. use samtools to determine proportion of reads that map
8. map short-read data
9. use samtools to look at coverage




