
# **Stalkie Reference Genome Assembly**
From [Alex's pipeline](https://github.com/alexjvr1/T.dalmanni_Genomics_of_meiotic_drive/blob/main/Electronic_Lab_Book.md#1-genome-assembly), the current _T. dalmanni_ ref is ~0.383 Gb and 3 chromosomes

### Samples & data
 Samples
  ~~~
  - 7 total, 2 per species (1M, 1F), plus 1 additional T. whitei F (potential driver).
  - Assemblies should be generated for each sample, so 7 assemblies total.
  - Two HiFi .fastq.gz files per individual (so 4,4,6) per species.
  ~~~

## **Genome Size Estimation with Jellyfish**
* [Jellyfish v2.2.10](https://github.com/gmarcais/Jellyfish)
* Ran [jellyfish.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/jellyfish.sh) on each species
* Used [genomescope](http://qb.cshl.edu/genomescope/) to visualise the .histo files generated by jellyfish (read length=10000, max kmer coverage=100000)
* [genomescope plots](https://github.com/BenAlston/stalkie_ref_genome_assembly/tree/main/lab_book/Data/jellyfish_histos%20)


# **Assembly with Hifiasm**
* [hifiasm v0.16.1-r375](https://github.com/chhylp123/hifiasm)
* Ran [hifiasm.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/hifiasm.sh) (Hifiasm V 0.16.1), takes ~10-12 hrs
* made hifiasm_hic.sh, for when hi-c arrives

### **Initial Hifiasm Assemblies**
* [output files documentation](https://hifiasm.readthedocs.io/en/latest/interpreting-output.html)
* duplication looks very high across the board, not very contiguous, also a lot larger than expected

**Table 1:** Assembly number corresponds with the identifier in the hifi read filenames (e.g., assembly 6 is derived from 200437_6-Cell1, which contains _T. dalmanni_ male reads). Est size from jellyfish.
| species | assembly | info         | est_size         | primary_assembly_size | busco_completeness | busco_dup | contigs | N50 (Kb) |
|-----|-----|-----------------|------------------|-----------------------|--------------------|-----------|---------|-----|
| whitei | 1 | female       |        597436869 | 761502157             | 95.8               | 35.6      | 7589    | 156 |
| whitei | 2 | male       | 424970908        | 730392625             | 95.1               | 25.6      | 6466    | 178 |
| whitei| 3 | female, potential driver |        584931232 | 731617842             | 95.7               | 42.3      | 6774    | 178 |
| meigenii| 4 | female      |        694112101 | 694692253             | 96.4               | 28.6      | 3136    | 463 |
| meigenii| 5 | male     | 477607885        | 742331081             | 96.3               | 16.9      | 3064    | 576 |
| meigenii| 5 | male, ran with -k 19 | 477607885 | 716000585 | 96.4 | 12.6 | 2816 | 587 |
| dalmanni| 6 | male       | 657136228        | 700361654             | 97.8               | 25.7      | 2913    | 687 |
|dalmanni | 7 | female       |        533280821 | 609130602             | 97.3               | 4.9       | 2425    | 998 |


### **BUSCO**
* Ran [busco.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/busco.sh), takes ~10-20 mins
* See [busco output](https://github.com/BenAlston/stalkie_ref_genome_assembly/tree/main/lab_book/Data/BUSCO_output)
* very high duplication across all species, completeness is ok

### **Checking Coverage is as Expected**
* Check hifiasm correctly identifies the homozygous and heterozygous peaks in the kmer plots, compare this to the jellyfish kmer plots as a sanity check.
  - This is true for meigenii_4, dalmanni_7
* Issue with meigenii_5 and dalmanni_6:  "peak_hom: 23; peak_het: -1"
  - this is a solvable issue, [see this thread](https://github.com/chhylp123/hifiasm/issues/245)
  - Potentially caused by the sample being highly heterozygous. Hifiasm only identifies one peak (the het peak) and misidentifies it as the hom peak. For meigenii_5, it should be "peak_het: 23, peak_hom: 46"
  - This is not consistent with est coverage (total read length/est genome size). However I don't trust the jellyfish est genome sizes
 * Can fix the -1 hom peak issue a few ways:

**Table 2:** Does default hifiasn identify the correct peak, and can this be fixed by running with the "-k 19" parameter
|assembly|hifiasm identifies correct peak?| Fixed with -k 19? |
|:----|:----|:---|
|whitei_1|na|na|
|whitei_2|na|na|
|whitei_3|na|na|
|meigenii_4|yes|na|
|meigenii_5|no|yes|
|dalmanni_6|no|running|
|dalmanni_7|yes|na|

**Log**
 * Ran hifiasm.sh with -D 10 - **did not solve the issue**
 * Ran with predicted size 0.476g (jellyfish predicted size) - **did not solve the issue**
 * Ran with --min-hist-cnt 6 - ignore counts below the number - **does not work** - output the same as standard
 * Ran with --purge-cov 46 - manually set hom peak - **does not work**, outdated 
 * Ran with option --hom-cov 46 - manually set hom peak - **does not work** - output the same as standard
 * running with -k 19 (kmer lenght) - works, hifiasm detects the peaks correctly in meigenii_5
   - Although i'm not sure what -k is actually doing
   - interestingly the new assembly is sligntly better than the old one, 
   

# Cleaning Assemblies
* Remove contamination using blobtoolkit, then remove duplicated haplotigs using purge_dups

## **Filtering Contamination with Blobtoolkit**
* [Blobtoolkit v3.1.6 ](https://github.com/blobtoolkit/blobtoolkit)
* Essentially takes blast output, coverage, and busco output, and filters the dataset for contamination
* Needs:
  - blast.out file (generated in [blast_par.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/tree/main/scripts/blast_par.sh))
  - coverage (generated in [blobtoolkit.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/edit/main/scripts/blobtoolkit.sh))
  - busco (generated in [busco.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/tree/main/scripts/busco.sh))
* the scripts are now up and running and ready to be used on any sample

### **Blast**

**Installing local blast database**
* Remote searches are not feasible for large queries, so installed the local Blast nt db ([using blast_nt_db.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/blast_nt_db.sh)), this takes up ~450GB of space
* Even with the local blast nt db, blasting my highly contiguous assemblies will take too long unless it is parallelised

**Parallelising Blast**
* split my assembly fasta into 500 smaller files: ran [fasta_splitter.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/fasta_splitter.sh)
* ran blast on each using an array script, [blast_par.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/tree/main/scripts/blast_par.sh)
* Then I condensed these blast output files into a single file:
~~~
cat out/* >> all_blast.out
~~~

## **Blobtoolkit pipeline:**
* Ran [blobtoolkit.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/edit/main/scripts/blobtoolkit.sh)
* This script calculates coverage, takes the blast output I just generated, and the busco output from [busco.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/edit/main/scripts/busco.sh). It then filters the assembly for contamination, I am able to specify which taxanomic rank blast matches I want removed.
* **<Important>** generating coverage using minimap2 does not work for dalmanni_6, this is likely an issue to to with the HiFi reads being to long.
* This analysis has currently been run on whitei_1 and whitei_2 only.

#### **Setting Filter Parameters**
* Need to select appropriate taxanomic level to filter
* The below table was generated with: --param bestsum_$clade_rank--keys=no-hit,$clade # this keeps only reads with blast hits from that clade, and reads with no blast hits
  - the filessizes and names for the lr mapping files seem off, im remapping eveything manually to see if it fixes anything
  - floating point error when generating calcuts, probably due to PB.stats containing only 0s
* need to decide on cutoffs - reread the forum post, get the script working (so i know the issues im having arent user error), then email the dev


| Clade      | Size (Gb) | Contigs | Completeness | Duplication|
|------------|-----------|---------|--------------|------------|
| Unfiltered | 0.762     | 7598    | 95.8         | 35.6       |
| Eukaryota  | 0.748     |  7481   | 95.7         | 35.6       |
| Metazoa    | 0.742     | 7368    | 95.6         | 35.5       |
| Arthropoda | 0.742     | 7363    | 95.6         | 35.5       |
| Diptera    | 0.715     | 6933    | 95.3         | 35.3       | 
* Metazoan and no blast matches seem to be the most sensible contigs to retain.

## **Purge Dups**
* [purge_dups v1.2.5](https://github.com/dfguan/purge_dups)
* Purges haplotigs using coverage and sequence similarity info
* First running this pipeline on whitei F (sample 1) since its busco dup is 35%

* Runing the pipeline manually to have more control over the parameters.

**Manual pipeline**

* Note changed -xasm20 to -x map-hifi, have not tested yet

whitei_1 pre and post purge_dups_manual.sh
|              | length (gb) | contigs | dup (%)| Conpleteness (%) | n50 (kb) |
| -------------|-------------|---------|--------|------------------|----------|
| pre-purging  | 0.762 | 7589   |  35.6|   95.8         | 156 |
| purge_dups round 2 (default cutoffs) | 0.548    | 4718      | 2.5    | 94.5     | 187 |


* Purge dups now runs, but ideally completeness should be higher, should probably manually tweak cutoffs.
* Have emailed the creator of purge_dups to ask about cutoffs:
![PB_1 cov](https://github.com/BenAlston/stalkie_ref_genome_assembly/assets/159305266/240949cb-0eae-4b67-a478-9ecd0d1e9c4a)
* These seem sensible, valley between het and hom peaks has been identified, but I am unsure if the upper and lower cutoffs are in the correct places

## **Sex Chromosome Identification**
* Differences in coverage and heterozygosity can be used to identify the XY chromosomes
* Illumina short read data for the three species, 5 individuals per sex

 **Options**
* Construct own pipeline
* Use [findZX](https://github.com/hsigeman/findZX)





