
# **Reference Genome Assembly**

[Useful resource](https://github.com/alexjvr1/T.dalmanni_Genomics_of_meiotic_drive/blob/main/Electronic_Lab_Book.md#1-genome-assembly)

## **1. Genome Size Estimation with Jellyfish**
* memory decided via hash size (size of the genome plus seq error) T. dalmanni genome is 0.62 gb, so i'll assume 0.7 (nanoplot doesnt accept less than 1G so do that)
* not sure what kmer size to use, 21 seems sensible 
* jellyfish V 2.2.10

Ran [jellyfish.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/jellyfish.sh) on each species

### **kmer distribution plots**
Used [genomescope](http://qb.cshl.edu/genomescope/) to visualise the .histo files generated by jellyfish (read length=10000, max kmer coverage=100000):
- [_T. whitei_ Female](http://genomescope.org/analysis.php?code=blR2SdZ6dlrwedM2Fgs8)
- [_T. whitei_ Female Driver](http://genomescope.org/analysis.php?code=VjP3GGnGS6YEajK6XUB1)
- [_T. whitei_ Male](http://genomescope.org/analysis.php?code=XWsvvvcObVy7x83qxEAu)
- [_T. dalmanni_ Female](http://genomescope.org/analysis.php?code=7NAz3STmNnNqW83R5kLc)
- [_T. dalmanni_ Male](http://genomescope.org/analysis.php?code=TW147FQXJ18A79EpLjEc)
- [_D. meigenii_ Female](http://genomescope.org/analysis.php?code=4tooUMZucAlqGmV0V0E8)
- [_D. meigenii Male_](http://genomescope.org/analysis.php?code=ISWwt723ucFqtJJKRgrT) 
<br><br>

| file prefix | sample         |     |
| ------------ | --------------- | ------ |
| 200437\_1    | whitei_F        |
| 200437\_2    | whitei_M        | Male   |
| 200437\_3    | whitei_F_driver | Female |
| 200437\_4    | meigenii_F      | Female |
| 200437\_5    | meigenii_M      | Male   |
| 200437\_6    | dalmanni_M     | Male   |
| 200437\_7    | dalmanni_F     | Female |

## **2. Assembly with Hifiasm**
* [Documentation](https://github.com/chhylp123/hifiasm)
* Ran [hifiasm.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/hifiasm.sh) (Hifiasm V 0.16.1), takes ~10-12 hrs

### Samples:

  ~~~
  - 7 total, 2 per species (1M, 1F), plus 1 additional T. whitei F (potential driver).
  - Assemblies should be generated for each sample, so 7 assemblies total.
  - Two HiFi .fastq.gz files per individual (so 4,4,6) per species.
  ~~~

| Sample ID | file\_prefix | Species         | Sex    |
| --------- | ------------ | --------------- | ------ |
| ST1       | 200437\_1    | C.whitei        | Female |
| ST2       | 200437\_2    | C.whitei        | Male   |
| ST3       | 200437\_3    | C.whitei Driver | Female |
| ST6       | 200437\_4    | D.meigenii      | Female |
| ST7       | 200437\_5    | D.meigenii      | Male   |
| ST8       | 200437\_6    | T.dalmanni      | Male   |
| ST9       | 200437\_7    | T.dalmanni      | Female |

### **Hifiasm output**
* [output files documentation](https://hifiasm.readthedocs.io/en/latest/interpreting-output.html)
* duplication looks very high across the board, not very contiguous


| sample          | est_size         | primary_assembly_size | busco_completeness | busco_dup | contigs |
|-----------------|------------------|-----------------------|--------------------|-----------|---------|
| whitei_f        |        597436869 | 761502157             | 95.8               | 35.6      | 7589    |
| whitei_m        | 424970908        | 730392625             | 95.1               | 25.6      | 6466    |
| whitei_f_driver |        584931232 | 731617842             | 95.7               | 42.3      | 6774    |
| dalmani_f       |        533280821 | 609130602             | 97.3               | 4.9       | 2425    |
| dalmani_m       | 657136228        | 700361654             | 97.8               | 25.7      | 2913    |
| meigenii_f      |        694112101 | 694692253             | 96.4               | 28.6      | 3136    |
| meigenii_m      | 477607885        | 742331081             | 96.3               | 16.9      | 3064    |


### **Inital QC with BUSCO**
* Used a docker/apptainer image
* ran [busco.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/busco.sh), takes ~10-20 mins
* see [busco output](https://github.com/BenAlston/stalkie_ref_genome_assembly/tree/main/lab_book/Data/BUSCO_output)
* very high duplication across all species, completeness is ok

### **Alignment With Miniap2**
* [Documentation](https://github.com/lh3/minimap2)

| Sample ID | aligned | Species         | Sex    |
| --------- | ------------ | --------------- | ------ |
| 1       | y    | C.whitei        | Female |
| 2       | n    | C.whitei        | Male   |
| 3       | n    | C.whitei Driver | Female |
| 4       | y    | D.meigenii      | Female |
| 5       | n    | D.meigenii      | Male   |
| 6       | y    | T.dalmanni      | Male   |
| 7       | n    | T.dalmanni      | Female |



## **Filtering with Blobtoolkit**
* [documentation](https://github.com/blobtoolkit/blobtoolkit)
* essentially takes blast output, coverage, and busco output, and filters the dataset for contamination. This program has been quite tricky to get working, considering it isn't doing anything that complext. This is probably because it does a bunch of other stuff that isn't relevant.

## **Ongoing:**
* BUSCO - done
* Blast Hits file - done for dalmanni
* coverage (long read remapped to primary assembly) - ongoing, running into issues, likely due to lack of memory
Error:
~~~
segmentation fault
~~~  
* Figured this was likley due to lack of memory so ran with 256 GB of ram and 32 cores
   Still fails with segmentation fault error
  - 83% core usage, only 12/256GB memory usage (though could be a sharp peak not recorded by the hpc)
  - All fail at: [M::worker_pipeline::1667.413*15.91] mapped 5521 sequences
  - ðŸ¤”
  - pretty sure I know what the issue is: most of the genomes are too repetative, this makes minimap2 upset
  - re run with whitei_3 and meigenni 5 to confirm 40 and 16% dup, respectivley.
  - meigenii_5 is 2267958, whitei_3 is the other one
  - the plot thickens, both test samples run fine, even whitei_3 at 40% duplication, dalmanni_6 is probably corrupted

**Summary:**
* Encontered errors in the mapping stage probably due to the high repeat content in the genome I was trying. It is probably worth reducing the duplication levels before continuing with this bit.


### **Blast**
* Remote searches take too long, so installing the Blast nt db
* only 500gb, installed using docker image [blast_nt_bd.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/tree/main/scripts/blast_nt_db.sh)
* Script ran overnight, has takes around an hour per seq, so would take tens of days for all 2400 contigs
* need to parralelize

Split assembly into 500 smaller fasta files:
~~~
module load Anaconda3/2022.05
source activate blast # conda env contains the package "fasta splitter"


ln -s /mnt/parscratch/users/bip23bta/ref_genomes/dalmanni/02-hifiasm/6_hifiasm_output/6_primary

# splits a given fasta file into 500 smaller ones
fasta-splitter --n-parts 500 $input_file --nopad --out-dir split

# count number of seqs in the output and make sure it sums to the same as original:
grep -c ">" out/6_primary.* | awk -F: '{total += $2} END {print "Total:", total}'
~~~


run blast on each using an array script, [blast_par.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/tree/main/scripts/blast_par.sh)

then condense into a single blast output file and remove temp files:
~~~
# when blast script has run, concentrate all outputs into one file
cat out/* >> all_blast.out
# should also write a script to remove the temp files
~~~

### **Alternatives to Blobtoolkit**
* [FCS-GX](https://github.com/ncbi/fcs/wiki/FCS-GX) is NCBIs contaminant removal tool. Looks good but requires installing a 470GB reference dataset.

## **Purge Dups**
* [Documentation](https://github.com/dfguan/purge_dups)
* First running this pipeline on whitei F (sample 1) since its busco dup is 35%
* generate config dir, ran [purge_dups_config.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/tree/main/scripts/purge_dups_config.sh)
  - seems to work, no error or output
  - manually modified the 1_config.json file to increase available memory (uses minimap2 so should need quite a bit)
* 


## **Next Steps:**
*  [findZX](https://github.com/hsigeman/findZX) has potential, look through the paper


1. Get blobtools working
3. use inspector for qc
4. extract mt genomes
7. use samtools to determine proportion of reads that map
8. map short-read data
9. use samtools to look at coverage




