
# **Genome Size Estimation with Jellyfish**
* memory decided via hash size (size of the genome plus seq error) T. dalmanni genome is 0.62 gb, so i'll assume 0.7 (nanoplot doesnt accept less than 1G so do that)
* not sure what kmer size to use, 21 seems sensible 
* jellyfish V 2.2.10

Ran [jellyfish.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/jellyfish.sh) on each species

### **kmer distribution plots**
Used [genomescope](http://qb.cshl.edu/genomescope/) to visualise the .histo files generated by jellyfish (read length=10000, max kmer coverage=1000000):
- [_T. whitei_ Female](http://genomescope.org/analysis.php?code=blR2SdZ6dlrwedM2Fgs8)
- [_T. whitei_ Female Driver](http://genomescope.org/analysis.php?code=VjP3GGnGS6YEajK6XUB1)
- [_T. whitei_ Male](http://genomescope.org/analysis.php?code=XWsvvvcObVy7x83qxEAu)
- [_T. dalmanni_ Female](http://genomescope.org/analysis.php?code=7NAz3STmNnNqW83R5kLc)
- [_T. dalmanni_ Male](http://genomescope.org/analysis.php?code=TW147FQXJ18A79EpLjEc)
- [_D. meigenii_ Female](http://genomescope.org/analysis.php?code=4tooUMZucAlqGmV0V0E8)
- [_D. meigenii Male_](http://genomescope.org/analysis.php?code=ISWwt723ucFqtJJKRgrT) 
<br><br>
* Some seem acceptable, containing two peaks, others e.g., _D. meigenii_ F seem bad, could be contamination.
* However, est. genome size seems odd, being variable, and larger in females than males in _D. meigenii_ and _T. whitei_ and the opposite in _T. dalmanni_

# **Assembly with Hifiasm**
* [Documentation](https://github.com/chhylp123/hifiasm)
* Samples:

  ~~~
  - 7 total, 2 per species (1M, 1F), plus 1 additional T. whitei F (potential driver).
  - Assemblies should be generated for each sample, so 7 assemblies total.
  - Two HiFi .fastq.gz files per individual (so 4,4,6) per species.
  ~~~

*  Ran [hifiasm.sh](https://github.com/BenAlston/stalkie_ref_genome_assembly/blob/main/scripts/hifiasm.sh) (Hifiasm V 0.16.1)
* Currently not working, was previously a memory issue, but now just seems stuck.

### **Hifiasm output**
* output files documentation
  - $prefix.bp.p_ctg.gfa: assembly graph of primary contigs.
  - $prefix.bp.hap1.p_ctg.gfa: partially phased contig graph of haplotype1.
  - $prefix.bp.hap2.p_ctg.gfa: partially phased contig graph of haplotype2.

## **Next Steps:**
* BUSCO
* Jellyfish
* purge_dups, if needed
